Sources/Plasma/PubUtilLib/plGImage/plFont.cpp

```diff
@@ -1128,7 +1128,7 @@ void    plFont::IRenderChar8To32AlphaPremShadow( const plFont::plCharacter &c )
     y = fRenderInfo.fClipRect.fY - fRenderInfo.fY + (int16_t)c.fBaseline;
     if( y < -2 )
         y = -2;
-    destBasePtr = (uint32_t *)( (uint8_t *)destBasePtr + y*fRenderInfo.fDestStride );
+    destBasePtr = (uint32_t *)((uint8_t *)destBasePtr + y * int32_t(fRenderInfo.fDestStride));
```

Comment:
Is this the only spot this happens, or does it do this same calculation for each of the different rendering types?

Reply:
It's the only one I found that had an explicit negative value, but I did find another potentially negative multiplier (just pushed)...  The rest are either already checked for negative or already converted to signed.

Reply:
IIRC This function was written by @cwalther to support text shadowing in the KI.

Reply:
Right – and I had probably just forgotten that int16_t * uint32_t = uint32_t at that point. The code was copied from other functions, but this is the only one where y can be negative (because shadows can extend beyond the top of the character).

What about line 367? A quick glance around line 266 suggests that `fRenderInfo.fY` could be negative. That would make `fDestPtr` go out of bounds, but further arithmetic might bring it back into bounds later before being dereferenced.